### PR TITLE
Don't use ObserveOn in Bindings.

### PR DESCRIPTION
**Pull request by grokys**
Instead of using `Observable.ObserveOn` in bindings, interface with `Dispatcher.UIThread` to schedule binding notifications on the UI thread. This saves a significant amount of memory.

Using the VS2017 memory profiler on ControlCatalog after cycling through each page I get the following results:

```
before: 68.96MB 2,764,678 objects
after: 57.07MB 2,357,602 objects
```

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]

**Comment by danwalmsley** [comment]
Wow! Avalonstudio was 151.5MB usage and with this PR its 79.6MB

File: src/Avalonia.Base/AvaloniaObject.cs [hunk 1]
```diff
@@ -340,7 +340,6 @@ public bool IsSet(AvaloniaProperty property)
             var description = GetDescription(source);
 
             var scheduler = AvaloniaLocator.Current.GetService<IScheduler>() ?? ImmediateScheduler.Instance;
```
Comment (kekekeks) [inline src/Avalonia.Base/AvaloniaObject.cs]: Variable is not used anywhere

File: src/Avalonia.Base/PriorityBindingEntry.cs [hunk 2]
```diff
@@ -92,33 +93,52 @@ public void Dispose()
 
         private void ValueChanged(object value)
         {
-            _owner.Owner.Owner?.VerifyAccess();
+            void Signal()
+            {
+                _owner.Owner.Owner?.VerifyAccess();
```
Comment (kekekeks) [inline src/Avalonia.Base/PriorityBindingEntry.cs]: Is it really needed to call VerifyAccess here? You've just checked it and scheduled the call back to the UI thread

**Comment by kekekeks** [comment]
Other than that LGTM

**Review by kekekeks — APPROVED** [review]

**Comment by grokys** [comment]
@danwalmsley that's amazing, I didn't expect that much difference!

@kekekeks yep, good catches. Removed those lines.

**Pull request merged**